### ISSUE TITLE
perf(spec): skip logits replicate_to_mesh in greedy verify path

### ISSUE THREAD

**Issue by lingebeng** [sgl-project/sglang#26328]
## Problem

During the verify phase of EAGLE3 speculative decoding, `replicate_to_mesh` copies `next_token_logits` from TP-sharded `P("data", "tensor")` to fully replicated `P()`.

For Qwen3-32B (vocab=152064, TP=4, bs=16), this means an all-gather of **~3.5MB** across cores on every verify iteration, showing up as a **7-8ms** `XlaLinearize` bubble in xprof traces.

<img width="1380" height="602" alt="Image" src="https://github.com/user-attachments/assets/291cd8b2-f772-448c-ab43-3787a1aeeeb7" />

## Analysis

The logits replicate is unnecessary in the greedy sampling path:

1. **`jnp.argmax` in `verify_tree_greedy`**: JAX SPMD compiler natively handles argmax on sharded data (per-shard local argmax + cross-shard all-reduce) — no upfront replicate needed.
2. **Downstream `logits_output.next_token_logits[safe_index, :]` gather**: The gathered logits are never consumed downstream in the EAGLE greedy path (`draft_extend_for_decode` only uses `hidden_states`).

Therefore, the greedy path only needs to replicate `hidden_states` (~30KB), not `next_token_logits` (~3.5MB).

## Solution

Branch on `is_all_greedy` in `EAGLEWorker.verify`:
- **Greedy**: replicate only `hidden_states`, skip logits replicate and gather
- **Non-greedy**: preserve existing behavior, replicate both

## Impact
bs = 16
|                                        | Data size | Replicate latency |
| -------------------------------------- | --------- | ----------------- |
| Before: logits (bs×3, 152064) bf16     | ~3.5MB    | 7-8ms             |
| After: hidden_states (bs×3, 5120) bf16 | ~30KB     | ~0.18ms           |

Eliminates **~7ms** of TPU idle bubble per verify iteration.

<img width="1106" height="440" alt="Image" src="https://github.com/user-attachments/assets/ac8c8cd0-aaec-4c14-a4ea-a62d0fba5f13" />

**Comment by lingebeng** [comment]
Sorry, it's my error. It should be in the sglang-jax project.
